features/facebookImport/src/model/analysis.js

```diff
@@ -255,5 +255,6 @@ export async function analyzeZip(zipData, zipFile, facebookAccount, pod) {
 
 export async function analyzeFile(zipData, facebookAccount) {
     const zipFile = new ZipFile(zipData, window.pod);
+    await zipFile.refreshCachedEntries();
```

Comment:
Uh, I think I had an idea I like even more: You could change `refreshCachedEntries()` a bit so that it only does anything if it's not _already_ cached. Probably give it a more fitting name, like `cacheEntries()` or something.

Then both `getEntries()` and `hasEntries()` could just call `this.cacheEntries()` before they do anything, but you could _also_ call it right after constructing `ZipFile` to ensure it's not being cached multiple times.

Still not super elegant, but at least callers don't have to remember to use `ZipFile.create` rather than the constructor, which is kinda error prone.

Reply:
I suppose we _could_ have a field called `_promisedEntries`, and basically assign the result of `polyOut.readdir` (a promise) to it, but also check if it's already there and if so, just `await` it. That should work without too much hassle, but might still be overkill at this time.

Reply:
Did a pass over this. There is `_ensureCachedEntries` and users do not have to call necessarily the static method